src/main/java/com/google/sps/servlets/NewRecipeServlet.java

```diff
@@ -25,13 +25,21 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String recipeName = request.getParameter("recipeName");
+    String name = request.getParameter("name");
     String tagsResponse = request.getParameter("tags");
     String description = request.getParameter("description");
     String ingredientsResponse = request.getParameter("ingredients");
     String stepsResponse = request.getParameter("steps");
     // privacy will be used once we give users the option to make their recipes public
-    String privacy = request.getParameter("privacy");
+    // String privacy = request.getParameter("privacy");
+
+    // prevents empty responses as being saved in datastore
+    // redirects user back to UserPage
+    // TODO: inform the user that they're missing stuff
+    if (name.equals("") || tagsResponse.equals("") || description.equals("") || ingredientsResponse.equals("") || stepsResponse.equals("")){
+      response.sendRedirect("/pages/UserPage.html");
```

Comment:
You added some redirecting here which wasn't in the last review. For now I say it looks good, but Ali and I have both had instances where this doesn't work as expected, so I would suggest double checking it behaves properly.

Reply:
I've played around with it and it seems to work but if it starts breaking out of nowhere I'll know to look here